pkg/internal/day/duration.go

```diff
@@ -68,8 +75,23 @@ func handleDays(s []byte) (time.Duration, error) {
 }
 
 // String returns a string value of DayDuration.
-func (d Duration) String() string {
-	return time.Duration(d).String()
+func (d *Duration) String() string {
+	return d.T.String()
+}
+
+// Text returns the original text received while parsing.
+func (d *Duration) Text() string {
+	return d.Txt
+}
+
+// Duration returns the parsed duration.
+func (d *Duration) Duration() time.Duration {
+	return d.T
+}
+
+// SetDuration returns the parsed duration.
+func (d *Duration) SetDuration(t time.Duration) {
```

Comment:
Also it feels that `Duration` should be immutable because it's derived from text input.